src/cls/journal/cls_journal.cc

```diff
@@ -540,10 +540,12 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
                             bufferlist *out) {
   std::string id;
   bufferlist data;
+  std::set<cls::journal::Client> clients;
   try {
     bufferlist::iterator iter = in->begin();
     ::decode(id, iter);
     ::decode(data, iter);
+    ::decode(clients, iter);
```

Comment:
It's still potentially stale data -- you can use the current cls_journal::journal_client_list method as a guide for how to do it within the class object (it will be simpler since you don't have to worry about start after/max return).